12-watchtower-API.md

```diff
@@ -127,122 +129,167 @@ This message contains all the information regarding the appointment that the cli
 	* MUST set `op_customer_signature` to the signature of the appointment using `op_customer_signature_algorithm`.
-	* MUST set `op_customer_public_key` the public key that matches the private key used to create `op_customer_signature`.
+	* MUST set `op_customer_public_key` to the public key that matches the private key used to create `op_customer_signature`.
 
 The receiving node:
+* upon receiving a `transaction_fee`:
+	* MUST compute the `fee_rate` set in the `justice_tx`.
 
-* upon receiving a `locator` that is not a `16-byte` value:
-	* MUST reject the appointment.
-* upon receiving a `start_block` behind the current chain tip:
-	* MUST reject the appointment.
-* upon receiving a `start_block` too close to the current chain tip:
-	* SHOULD reject the appointment.
-* upon receiving a `start_block` too far away in the future:
-	* SHOULD reject the appointment.
-* upon receiving an `end_block` behind the current chain tip:
-	* MUST reject the appointment.
-* upon receiving an `end_block` too far away in the future:
-	* SHOULD reject the appointment.
-* upon receiving a `dispute_delta` too small:
-	* SHOULD reject the appointment.
-* upon receiving an `encrypted_blob` of non-feasible size:
-	* MUST reject the appointment.
-* upon receiving a `cipher` that he does not implement:
-	* MUST reject the appointment. 
-* upon receiving a `transaction_size` too big:
-	* MUST reject the appointment.
-* upon receiving a `transaction_size` too small:
-	* MUST reject the appointment.
-* upon receiving a `transaction_fee` too low:
-	* SHOULD reject the appointment.
+The receiving node MUST reject the appointment if:
 
-* if `qos` was agreed on `wt_init`:
-	* if `op_customer_signature_algorithm` is missing:
-		* MUST reject the appointment.
-	* if `op_customer_signature_algorithm` does not match with one of the supported signing algorithms:
-		* MUST reject the appointment.
-	* if `op_customer_signature` is missing:
-		* MUST reject the appointment.
-	* if `op_customer_public_key` is missing:
-		* MUST reject the appopintment.
-	* if `op_customer_signature` cannot be verified using `op_customer_public_key`:
-		* MUST reject the appopintment.
-
-* if `qos` was not agreed on `wt_init`:
-	* if `op_customer_signature_algorithm` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_signature` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_public_key` is present:
-		* SHOULD reject the appopintment.
+* `locator` is not a `16-byte` value.
+* `start_block` is not an integer.
+* `start_block` is behind the current chain tip.
+* `end_block` is not an integer.
+* `end_block` is behind the current chain tip.
+* `dispute_delta` is not an integer.
+* `encrypted_blob` has non-feasible size.
+* `cipher` is not among the one he implements.
+* `transaction_size` is non-feasible.
 
+The receiving node SHOULD reject the appointment if:
+
+* `start_block` is too close to the current chain tip.
```

Comment:
why?

Reply:
it is likely we will get appointments based on the current chain tip. that is ok considering the end_block should be far away.

Reply:
See previous comment regarding the same thing